src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+
+import { Loader } from '../Loader';
+import { PeopleTable } from '../Loader/PeopleTable';
+import { getPeople } from '../services/api';
+import { Person } from '../../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then(data => {
+        const peopleByName: { [key: string]: Person } = {};
+
+        data.forEach(person => (peopleByName[person.name] = person));
+        const dataUpdated = data.map(p => {
+          return {
+            ...p,
+            mother: peopleByName[p.motherName || ''],
+            father: peopleByName[p.fatherName || ''],
+          };
+        });
+
+        setPeople(dataUpdated);
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
```

Comment:
This is a code quality issue. You're setting the loading state to false only after the data has been fetched. If an error occurs, the user will be stuck with the loading spinner. Consider setting the loading state to false in a finally block to ensure it is always set to false, regardless of whether or not an error occurred.